### DISCUSSION TITLE
Video support error

### DISCUSSION THREAD

**Discussion by wizgod** [bpatrik/pigallery2#817] [General]
Hi @bpatrik,

After switching to the latest version, I am running into the following error with videos and ffmpeg where I never had an issue with the previous version.

(1) Video support error, switching off..
"Error: Error accessing ffmpeg, cant find executable: Error: spawn /public_html/node_modules/ffmpeg-static/ffmpeg EACCES"

First I thought that "spawn" was a package I needed installed but an "apt install" didn't find it.

Any thoughts?

Thanks,

Wg

**Comment by martadinata666** [comment]
For some reason it cant execute ffmpeg due permission issue.

**Reply by wizgod** [reply]
Hi @martadinata666,

The permissions on ffmpeg-static are the same as the previous version - 664 (same ownership).

I did change it to 755 to see if that would work but still got the error.

I also restarted from scratch and did an npm install with the same result (I confirmed the ffmpeg folders were there).

Here is the detailed error:

`ErrorDTO {
  code: 60,
  message: 'Settings error: Error: Error accessing ffmpeg, cant find executable: Error: spawn /home/wizgod/domains/wizgod.gallery/public_html/node_modules/ffmpeg-static/ffmpeg EACCES',
  details: Error: Error accessing ffmpeg, cant find executable: Error: spawn /home/wizgod/domains/wizgod.gallery/public_html/node_modules/ffmpeg-static/ffmpeg EACCES
      at /home/wizgod/domains/wizgod.gallery/public_html/src/backend/model/diagnostics/ConfigDiagnostics.js:79:43
      at /home/wizgod/domains/wizgod.gallery/public_html/node_modules/fluent-ffmpeg/lib/capabilities.js:360:16
      at ChildProcess.<anonymous> (/home/wizgod/domains/wizgod.gallery/public_html/node_modules/fluent-ffmpeg/lib/processor.js:159:9)
      at ChildProcess.emit (node:events:514:28)
      at ChildProcess._handle.onexit (node:internal/child_process:289:12)
      at onErrorNT (node:internal/child_process:476:16)
      at process.processTicksAndRejections (node:internal/process/task_queues:82:2                               1),
  request: { method: '', url: '' }
}
`

Thanks,

Wg

**Reply by martadinata666** [reply]
Is the file there? Is it can manually executed?

**Reply by wizgod** [reply]
Hi @martadinata666,

The file is there and I tried manually and received a permission denied error.

I changed the permissions to 775 and am now receiving a "Segmentation fault" error:

ErrorDTO {
  code: 60,
  message: 'Settings error: Error: Error accessing ffmpeg, cant find executable: Error: ffmpeg was killed with signal SIGSEGV',
  details: Error: Error accessing ffmpeg, cant find executable: Error: ffmpeg was killed with signal SIGSEGV
      at /home/wizgod/domains/wizgod.gallery/public_html/src/backend/model/diagnostics/ConfigDiagnostics.js:79:43
      at /home/wizgod/domains/wizgod.gallery/public_html/node_modules/fluent-ffmpeg/lib/capabilities.js:360:16
      at handleExit (/home/wizgod/domains/wizgod.gallery/public_html/node_modules/fluent-ffmpeg/lib/processor.js:170:11)
      at Socket.<anonymous> (/home/wizgod/domains/wizgod.gallery/public_html/node_modules/fluent-ffmpeg/lib/processor.js:197:11)
      at Socket.emit (node:events:526:35)
      at Pipe.<anonymous> (node:net:323:12),
  request: { method: '', url: '' }

Thanks,

Wg

**Reply by martadinata666** [reply]
Can you execute this? `/usr/bin/ffmpeg`? or at least install `ffmpeg`. Is this docker or manual install?

**Reply by wizgod** [reply]
Hi @martadinata666,

Yes, ffmpeg is installed and I can execute it from /usr/bin/ffmpeg and it's a manual install.

Thinking that maybe I messed up the environment, I reverted to 1.9.6 and didn't have the issue.

Thanks,

Wg

**Reply by wizgod** [reply]
What I did end up doing is using the node_modules folder from 1.9.6 and running `npm install` to install the missing packages for 2.0.3 and I no longer receive the error.

I'm going to speculate that the issue was with the version of one of the packages when I did a clean `npm install`.

**Reply by martadinata666** [reply]
I rebuild to "latest" master. Then yes there is a video playback issue but not ffmpeg EACESS but https://github.com/bpatrik/pigallery2/issues/808, it just refuses to play videos. Maybe related.

**Reply by wizgod** [reply]
I built the latest just now but still used the 1.9.6 node_modules folder and I had no video playback issues.

If I did a clean `npm install` then I ran into the issues I've mentioned previously.

**Reply by wizgod** [reply]
Hi @martadinata666,

It seems I was mistaken about the video playback. The video I played on 2.0 was the same I played on 1.9 so it was probably playing from cache.

Now I just tried to play other videos and they wouldn't play (including the original I tested with) - I probably should have done a CTRL-F5 before testing the first time. 

Unlike in #808, there is no error or any debug output - it shows the video player and the play button but does not start the video automatically nor does it do anything when clicking either play buttons. 

Wg

**Comment by wizgod** [comment]
It seems that after everything attempted, a final clean install got rid of the error (I have no clue as to what changed other than what I've mentioned previously).

I can see video thumbnails but not play them - something for another discussion.

**Reply by martadinata666** [reply]
In my deployment something like this 
![2024-01-23-11:52:20-screenshot](https://github.com/bpatrik/pigallery2/assets/2433562/88196d06-4941-4e89-9a66-3b33550a943b)

So if the video "thumbnail" is larger than the text will be overlayed.

**Reply by wizgod** [reply]
Here's what I see - this is using latest source, nothing modified, just video added.


https://github.com/bpatrik/pigallery2/assets/7222426/d755c733-5627-447c-a30b-683509c1a080

**Comment by wizgod** [comment]
Greetings Programs!

I finally had a chance to switch to v2.0.3 and I am no longer experiencing the error and video playback issue I was experiencing with 2.0.0.

Thanks,

Wg
